x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -1065,6 +1091,57 @@ class TimeseriesChartIntl extends Component {
     cxtGroup.append('path').datum(data).attr('class', 'values-line').attr('d', contextValuesLine);
     drawLineChartDots(data, cxtGroup, contextValuesLine, 1);
 
+    // Add annotation markers to the context area
+    cxtGroup.append('g').classed('mlContextAnnotations', true);
+
+    const [contextXRangeStart, contextXRangeEnd] = this.contextXScale.range();
+    const ctxAnnotations = cxtGroup
+      .select('.mlContextAnnotations')
+      .selectAll('g.mlContextAnnotation')
+      .data(showAnnotations && annotationData ? annotationData : [], (d) => d._id || '');
+
+    ctxAnnotations.enter().append('g').classed('mlContextAnnotation', true);
+
+    const ctxAnnotationRects = ctxAnnotations
+      .selectAll('.mlContextAnnotationRect')
+      .data((d) => [d]);
+
+    ctxAnnotationRects
+      .enter()
+      .append('rect')
+      .attr('rx', 2)
+      .attr('ry', 2)
+      .classed('mlContextAnnotationRect', true);
+
+    ctxAnnotationRects
+      .attr('x', (d) => {
+        const date = moment(d.timestamp);
+        let xPos = this.contextXScale(date);
+
+        if (xPos - ANNOTATION_SYMBOL_HEIGHT <= contextXRangeStart) {
+          xPos = 0;
+        }
+        if (xPos + ANNOTATION_SYMBOL_HEIGHT >= contextXRangeEnd) {
+          xPos = contextXRangeEnd - ANNOTATION_SYMBOL_HEIGHT;
+        }
+
+        return xPos;
+      })
+      .attr('y', cxtChartHeight + swlHeight + 2)
+      .attr('height', ANNOTATION_SYMBOL_HEIGHT)
+      .attr('width', (d) => {
+        const s = this.contextXScale(moment(d.timestamp)) + 1;
+        const e =
```

Comment:
shall we call these variables `start` and `end` instead?

Reply:
Renamed here https://github.com/elastic/kibana/pull/81490/commits/e84846f86e82dc04e7084cd82fc204aab01f9b98